packages/dts-generator/lib/phases/ast-transform.js

```diff
@@ -215,6 +216,47 @@ function addDefineArrayInterface(ast, symbolTable) {
   }
 }
 
+/**
+ * Adds generic type to `extend` method and modify `oClassInfo` type.
+ * This is done to specify `this` type inside the `oClassInfo` methods.
+ *
+ * @param symbolTable
+ */
+function modifyExtendMethods(symbolTable) {
+  _.forEach(symbolTable, (symbol) => {
+    if (symbol.kind !== "Class") {
+      return;
+    }
+
+    const extendMethod = _.find(
+      symbol.methods,
+      (method) => method.name === "extend"
```

Comment:
we want to be super strict here to avoid modifying other (similar) methods.
Lets also check the method is **static**?